src/tools/compiletest/src/common.rs

```diff
@@ -8,14 +8,14 @@ use camino::{Utf8Path, Utf8PathBuf};
 use semver::Version;
 use serde::de::{Deserialize, Deserializer, Error as _};
 
-pub use self::Mode::*;
+pub use self::TestMode::*;
 use crate::executor::{ColorConfig, OutputFormat};
 use crate::fatal;
 use crate::util::{Utf8PathBufExt, add_dylib_path, string_enum};
 
 string_enum! {
     #[derive(Clone, Copy, PartialEq, Debug)]
-    pub enum Mode {
+    pub enum TestMode {
```

Comment:
How does `TestMode` differ from a "test suite"? One mode maps to 1-N test suites? And a test suites is essentially just a directory within `tests`?

Reply:
It's *roughly* intended to be a 1-to-N mapping, i.e. each test mode can correspond to multiple test suites.

![compiletest-test-mode](https://github.com/user-attachments/assets/d933b1b9-518b-4766-ac12-3397a62ab58d)

Reply:
However, note that `tests/coverage/` is a special case: it is `TestSuite::Coverage`, but the same test files can be run under two test modes, `TestMode::{CoverageMap,CoverageRun}`.